WordPress/Sniffs/Theme/NoTitleTagSniff.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of the <title> tag, unless it is within a <svg> tag.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoTitleTagSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Property to keep track of whether a <svg> open tag has been encountered.
+	 *
+	 * @var bool
+	 */
+	private $in_svg = false;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_INLINE_HTML, // Finds inline html.
+			T_CONSTANT_ENCAPSED_STRING, // Finds the title tag in PHP.
+		 );
+	} // end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token in
+	 *                                        the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens  = $phpcsFile->getTokens();
+		$content = $tokens[ $stackPtr ]['content'];
+
+		// No need to check an empty string.
+		if ( '' === trim( $content ) ) {
+			return;
+		}
+
+		// Are we in a <svg> tag ?
+		if ( true === $this->in_svg ) {
+			if ( false === strpos( $content, '</svg>' ) ) {
+				// Rest in_svg at the end of the file.
+				if ( ( $phpcsFile->numTokens - 1 ) === $stackPtr ) {
```

Comment:
Ok, I'll try to explain better as it will *not* work.

```php
<?php
// file 1
echo 'some string<svg>';
my_function();
?>

<?php
// file 2
$var = 'abc';
?>
<p>some inline <title>fake title</title>html</p>
<?php
another_function_call();
?>
```

The sniff is listening for the `T_CONSTANT_ENCAPSED_STRING` and `T_INLINE_HTML` tokens, so the `process()` method will only be called for `'some string<svg>'` in file 1 and for `'abc'` and `<p>some inline <title>fake title</title>html</p>` for file 2.

The method will *not* be called for the last token in the file which is not a string, so comparing the `$stackPtr` to the token index for the last token in the file will not work *unless* the last token in the file is one of the two types of tokens you are listening for - which it isn't in either of the two example files above.

So `process()` gets called for `'some string<svg>'` in file 1 and `$this->in_svg` will be set to true.
Then `process()` gets called for `'abc'` in file 2. This will not even be examined as `$this->in_svg` is *still* set to `true`.
Next `process()` gets called for `<p>some inline <title>fake title</title>html</p>` and again, it will not be examined because `$this->in_svg` is still set to `true` and while this *should* be an error, no error will be thrown.

Does this make it clearer ?